gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/step/map/GroupStep.java

```diff
@@ -79,6 +79,14 @@ public void modulateBy(final Traversal.Admin<?, ?> kvTraversal) {
     }
 
     @Override
+    public void replaceLocalChild(final Traversal.Admin<?, ?> oldTraversal, final Traversal.Admin<?, ?> newTraversal) {
+        if (null != this.keyTraversal && this.keyTraversal.equals(oldTraversal))
+            this.keyTraversal = this.integrateChild(newTraversal);
+        else if (null != this.valueTraversal && this.valueTraversal.equals(oldTraversal))
```

Comment:
You are thinking of your strategy only and not about the general semantics of `replaceLocalChild()`. Your additions to `TraversalParent` should not just be predicated on your specific `ByModulatorStrategy`.